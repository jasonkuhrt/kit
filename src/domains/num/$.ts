/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
