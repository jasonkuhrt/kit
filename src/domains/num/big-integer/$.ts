export * as BigInteger from './$$.ts'
