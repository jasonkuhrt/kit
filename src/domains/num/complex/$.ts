export * as Complex from './$$.js'
