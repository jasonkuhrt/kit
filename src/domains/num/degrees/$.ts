// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
