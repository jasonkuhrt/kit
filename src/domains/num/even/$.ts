// Export the namespace containing all Even operations
export * as Even from './$$.ts'
