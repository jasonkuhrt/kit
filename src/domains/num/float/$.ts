// Export the namespace containing all Float operations
export * as Float from './$$.js'
