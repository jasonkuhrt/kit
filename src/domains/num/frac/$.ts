// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
