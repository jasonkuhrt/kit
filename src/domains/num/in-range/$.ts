// Export the namespace containing all InRange operations
export * as InRange from './$$.js'
