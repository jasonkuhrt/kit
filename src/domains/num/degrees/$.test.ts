import { Test } from '#test'
import { Degrees } from './$.js'

Test.describe('is')
  .on(Degrees.is)
  .casesInput(
    // Valid cases - any finite number is a valid degree
    [0],
    [90],
    [180],
    [360],
    [-90],
    [45.5],
    // Invalid cases - non-finite numbers
    [Infinity],
    [-Infinity],
    [NaN],
    ['90'],
    [null],
    [undefined],
  )
  .test()

Test.describe('from')
  .on(Degrees.from)
  .cases(
    // Valid cases
    [0],
    [90],
    [180],
    [360],
    [-90],
    [45.5],
    // Invalid cases - should throw
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()

Test.describe('tryFrom')
  .on(Degrees.tryFrom)
  .cases(
    // Valid cases
    [0],
    [90],
    [180],
    [360],
    [-90],
    [45.5],
    // Invalid cases - should return null
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()
