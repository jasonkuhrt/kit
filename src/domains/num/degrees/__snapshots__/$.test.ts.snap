// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-90
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-90
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-Infinity
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: -Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(45.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
45.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
45.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
90
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
90
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(180) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
180
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
180
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(360) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
360
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
360
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
Infinity
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(NaN) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
NaN
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is('90') 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
90
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-90
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-Infinity
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(45.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
45.5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
90
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(180) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
180
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(360) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
360
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
Infinity
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(NaN) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
NaN
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-90
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-90
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-Infinity
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(45.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
45.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
45.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(90) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
90
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
90
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(180) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
180
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
180
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(360) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
360
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
360
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(Infinity) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
Infinity
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(NaN) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
NaN
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;
